src/modules/tools.ts

```diff
@@ -37,3 +38,163 @@ export function getCharacterSet (label: string): CharacterSet {
+    return [{
+      trait_type: '3D',
+      value: '0x999'
+    }]
+  }
+
+  // 4D account
+  if (name.length === 4 && /^\d+$/.test(name)) {
+    const category = { trait_type: '4D', value: '' }
+    category.value = '10k club'
+    categories.push(category)
+
+    const values = ['AAAA', 'ABBB', 'AAAB', 'AABA', 'ABAA', 'AABB', 'ABAB', 'ABBA', 'AABC', 'ABCC']
+    const value = getReduplicationValue(values, name)
+    if (value !== '') {
+      categories.push({
+        trait_type: '4D',
+        value
+      })
+    }
+
+    // MMDD: month and day
+    const month = name.slice(0, 2)
+    const day = name.slice(2)
+    if (month >= '01' && month <= '12' && day >= '01' && day <= '31') {
+      categories.push({
+        trait_type: '4D',
+        value: 'MMDD'
+      })
+    }
+
+    // ABCD
+    if (isContinus(name, false)) {
+      categories.push({
+        trait_type: '4D',
+        value: 'ABCD'
+      })
+    }
+
+    // DCBA
+    if (isContinus(name, true)) {
+      categories.push({
+        trait_type: '4D',
+        value: 'DCBA'
+      })
+    }
+    return categories
+  }
+
+  if (name.length === 6 && name.startsWith('0x') && /^\d+$/.test(name.slice(2))) {
+    return [{
+      trait_type: '4D',
+      value: '0x10k'
+    }]
+  }
+
+  // 5D account
+  if (name.length === 5 && /^\d+$/.test(name)) {
+    const category = { trait_type: '5D', value: '100k club' }
+    categories.push(category)
+    const values = ['AAAAA', 'AABAA', 'ABBBB', 'AAAAB', 'ABAAA', 'AAABB', 'AABBB', 'AAABA', 'ABBBA', 'ABCCC', 'ABBBC', 'AAABC', 'ABABA', 'ABCBA']
+    const value = getReduplicationValue(values, name)
+    if (value !== '') {
+      categories.push({
+        trait_type: '5D',
+        value
+      })
+    }
+
+    // ABCDE
+    if (isContinus(name, false)) {
+      categories.push({
+        trait_type: '5D',
+        value: 'ABCDE'
+      })
+    }
+
+    // EDCBA
+    if (isContinus(name, true)) {
+      categories.push({
+        trait_type: '5D',
+        value: 'EDCBA'
+      })
+    }
+    return categories
+  }
+
+  if (name.length === 7 && name.startsWith('0x') && /^\d+$/.test(name.slice(2))) {
+    return [{
+      trait_type: '5D',
+      value: '0x100k'
+    }]
+  }
+
+  // Letter
+  if (/^[a-zA-Z]+$/.test(name)) {
+    const values = ['AAAA', 'ABBBA', 'AABB', 'ABAB', 'ABABA']
+    const value = getReduplicationValue(values, name)
+    if (value !== '') {
+      categories.push({
+        trait_type: 'Letter',
+        value
+      })
+    }
+    if (wordList.includes(name)) {
+      categories.push({
+        trait_type: 'Letter',
+        value: `Word${name.length}L`
+      })
+    }
+    return categories
+  }
+  return []
+}
+
+function isContinus (str: string, descend: boolean): boolean {
+  for (let i = 1; i < str.length; i++) {
+    if (str.charCodeAt(i) !== (str.charCodeAt(i - 1) + (descend ? -1 : 1))) {
+      return false
+    }
+  }
+  return true
+}
+
+function getReduplicationValue (values: string[], name: string) {
```

Comment:
Great.
But Named Capture Group is not a high-frequency feature, and the compatibility seems not high.

https://caniuse.com/?search=named%20capture

So there should be a link to the tech spec for the convenience of other developers.
https://github.com/tc39/proposal-regexp-named-groups